### PR TITLE
kmesh security: pod manage

### PR DESCRIPTION
**Pull request by lec-bit**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
kmesh secutity only manage pod managed by kmesh
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/workload/workload_processor.go [hunk 1]
```diff
@@ -104,12 +108,25 @@ func (p *Processor) getIdentityByUid(workloadUid string) string {
...
-	if workload.Node == p.nodeName {
+	podName := workload.Name
+	podNamespace := workload.Namespace
+	client, err := utils.GetK8sclient()
+	if err != nil {
+		log.Errorf("failed to get K8sclient, err: %v", err)
+		return false
+	}
+
+	pod, err := client.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_processor.go:119]: donot call apiserver each time, we could get from the informer cache

File: pkg/controller/workload/workload_processor.go [hunk 2]
```diff
@@ -104,12 +108,25 @@ func (p *Processor) getIdentityByUid(workloadUid string) string {
...
+	if err != nil {
+		log.Errorf("failed to get pod: %v", err)
+		return false
+	}
+
+	if pod.Annotations["kmesh.net/redirection"] == "enabled" && pod.Annotations["kmesh.net/bypass"] == "" {
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_processor.go:125]: use predefined constants

File: pkg/controller/workload/workload_processor.go [hunk 3]
```diff
@@ -104,12 +108,25 @@ func (p *Processor) getIdentityByUid(workloadUid string) string {
...
+	podNamespace := workload.Namespace
+	client, err := utils.GetK8sclient()
+	if err != nil {
+		log.Errorf("failed to get K8sclient, err: %v", err)
+		return false
+	}
+
+	pod, err := client.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("failed to get pod: %v", err)
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/workload_processor.go:121]:
```suggestion
		log.Errorf("get pod failed: %v", err)
```

**Review by hzxuzhonghu — DISMISSED** [review]

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -58,12 +58,21 @@ func NewController(opts *options.BootstrapConfigs, bpfWorkloadObj *bpf.BpfKmeshW
 }
 
 func (c *Controller) Start(stopCh <-chan struct{}) error {
+	var secertManager *security.SecretManager
+	var err error
+	if c.enableSecretManager {
+		secertManager, err = security.NewSecretManager()
```
[the remark this answers is not in the document]
Comment (lec-bit) [reply pkg/controller/controller.go:64]: no, ads mode is not support now

File: pkg/controller/manage/kmesh_manage.go [hunk 5]
```diff
@@ -23,6 +23,8 @@ import (
 	"strings"
 	"time"
 
+	"istio.io/istio/pkg/spiffe"
+
```
Comment (hzxuzhonghu) [inline pkg/controller/manage/kmesh_manage.go:27]: rm blank line

File: pkg/controller/manage/kmesh_manage.go [hunk 6]
```diff
@@ -133,6 +132,18 @@ func NewKmeshManageController(client kubernetes.Interface) (*KmeshManageControll
 				}
 
 				queue.AddRateLimited(QueueItem{pod: newPod, action: ActionDeleteAnnotation})
+				sendCertRequest(security, oldPod, kmeshsecurity.DELETE)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod, ok := obj.(*corev1.Pod)
+			if !ok {
```
Comment (hzxuzhonghu) [inline pkg/controller/manage/kmesh_manage.go:140]: To be more accurate add the below cast
```
	// When a delete is dropped, the relist will notice a job in the store not
	// in the list, leading to the insertion of a tombstone object which contains
	// the deleted key/value. Note that this value might be stale.
	if !ok {
		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
			return
		}
		job, ok = tombstone.Obj.(*corev1.Pod)
		if !ok {
			utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a Job %#v", obj))
			return
		}
	}
```

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**